libs/contrib/Data/List/Equalities.idr

```diff
@@ -0,0 +1,60 @@
+module Data.List.Equalities
+
+import Data.List
+
+%default total
+
+||| (::) is injective
+export
+consInjective : {x : a} -> {xs : List a} -> {y : b} -> {ys : List b} ->
+                (x :: xs) = (y :: ys) -> (x = y, xs = ys)
+consInjective Refl = (Refl, Refl)
+
+||| Two lists are equal, if their heads are equal and their tails are equal.
+consCong2 : {x : a} -> {xs : List a} -> {y : b} -> {ys : List b} ->
+            x = y -> xs = ys -> x :: xs = y :: ys
+consCong2 Refl Refl = Refl
+
+||| Equal non-empty lists should result in equal componens after destructuring 'snoc'.
+export
+snocCong2 : {x : a} -> {xs : List a} -> {y : b} -> {ys : List b} ->
```

Comment:
I would argue this is a good opportunity to add `snoc` as a
definition (in `Data.List`, I guess?).

Reply:
Do you mean to suggest something like:

```idris
snoc : List a -> a -> List a
snoc [] x = [x]
snoc (y :: xs) x = y :: snoc xs x
```

Reply:
I would be happy with it just being an alias for `xs ++ [x]`.

Reply:
Can we use the `snoc` definition in these theorems
now that it has been added?

Reply:
yes. updated PR.